File: backend/app/schemas/academic_year.py
from datetime import datetime

from pydantic import BaseModel, ConfigDict


class AcademicYearBase(BaseModel):
    year: int
    is_active: bool = True


class AcademicYearCreate(BaseModel):
    year: int


class AcademicYearRead(AcademicYearBase):
    id: int
    created_at: datetime
    updated_at: datetime

    model_config = ConfigDict(from_attributes=True)


class AcademicYearUpdate(BaseModel):
    year: int | None = None
    is_active: bool | None = None

File: frontend/src/types/academicYear.ts
// Mirrors backend/app/schemas/academic_year.py in full.
//
// Not explicitly listed in Task 32's scope, but added alongside it: a
// Semester only carries academic_year_id, not the year itself, and the
// task's own suggested ScheduleVersion label format ("{學期年份} 學期
// {semester_number} - ...") needs the actual year. Same five-set pattern
// as every other resource module, so this is a small, low-risk addition,
// not a scope change -- see Task 32's report for the explicit disclosure.

export interface AcademicYear {
  id: number;
  year: number;
  is_active: boolean;
  created_at: string;
  updated_at: string;
}

export interface AcademicYearCreate {
  year: number;
}

export interface AcademicYearUpdate {
  year?: number;
  is_active?: boolean;
}
